src/store/chuckApi/actions.js

```diff
@@ -0,0 +1,151 @@
+        category: `https://api.chucknorris.io/jokes/random?category=${searchCategory}`,
+        byInput: `https://api.chucknorris.io/jokes/search?query=${searchByInput}`
+    }
+    const isRandom = Boolean(!searchCategory && !searchByInput);
+    const isByInput = Boolean(searchByInput && !searchCategory);
+
+    const queryUrl = isRandom ? url.random : isByInput ? url.byInput : url.category
+    fetch(queryUrl)
+        .then((response) => response.json())
+        .then((data) => {
+            const isError = Boolean(!!data.total === false && !!data.id === false && !!data.error === true);
+            if (!isError) {
+                const checkedData = checkInFavoriteList(data, dispatch, getState)
+                return (dispatch(setJokeFromApi([])), dispatch(setJokeFromApi(checkedData)))
+            }
+        })
+}
+
+export const checkInFavoriteList = (results, dispatch, getState) => {
+    const state = getState();
+    const favoriteJokeList = state.chuckApi.favoriteJokeList;
+
+    const isResultsResult = Boolean(results.result !== undefined);
+    const correctResults = isResultsResult === true ? results.result : [results];
+
+    const modifiedResults = correctResults.map((item) => {
+        const { categories, icon_url, id, updated_at, url, value } = item;
+        const modifiedResultsItem = {
+            categories: !categories ? '' : categories,
+            icon_url: !icon_url ? '' : icon_url,
+            id: id,
+            updated_at: updated_at,
+            url: url,
+            value: value,
+            inFavorite: false
+        }
+        return modifiedResultsItem
+    })
+    const checkInStateFavoriteList = (item) => {
+        const foundJoke = favoriteJokeList.find(jokes => jokes.id === item.id)
+        return Boolean(foundJoke) === true ? foundJoke : item;
+    }
+    const isFavoriteJokeList = Boolean(favoriteJokeList === null);
+
+    return isFavoriteJokeList ? modifiedResults :
+        (dispatch(setFavoriteJokeList(favoriteJokeList)),
+            modifiedResults.map((item) => checkInStateFavoriteList(item)))
+}
+
+export const dispathAll = (jokeList, favoriteJokeList) => (dispatch) => {
+    return (localStorage.setItem('favoriteList', JSON.stringify(favoriteJokeList)),
+        dispatch(setJokeFromApi([])),
+        dispatch(setJokeFromApi(jokeList)),
+        dispatch(setFavoriteJokeList([])),
+        dispatch(setFavoriteJokeList(favoriteJokeList)))
+}
+
+export const addToFavoriteList = (categories, icon_url, id, updated_at, url, value) => (dispatch, getState) => {
+    const state = getState();
+    const favoriteJokeList = state.chuckApi.favoriteJokeList;
+    const jokeList = state.chuckApi.jokeList;
+
+    const isExistInFavoriteList = Boolean(favoriteJokeList.find(joke => joke.id === id));
+    const isExistInJokeList = Boolean(jokeList.find(joke => joke.id === id));
+
+    const isExistInBothList = isExistInJokeList === true && isExistInFavoriteList === true;
```

Comment:
For example here you can write just `isExistInJokeList && isExistInFavoriteList` because you already know that ther are both `boolean`